internal/processing/processors/migration/processors.go

```diff
@@ -0,0 +1,49 @@
+package migration
+
+import (
+	"github.com/go-logr/logr"
+	gatewayv1beta1 "github.com/kyma-project/api-gateway/apis/gateway/v1beta1"
+	gatewayv2alpha1 "github.com/kyma-project/api-gateway/apis/gateway/v2alpha1"
+	"github.com/kyma-project/api-gateway/internal/processing"
+	"github.com/kyma-project/api-gateway/internal/processing/processors/istio"
+)
+
+// NewMigrationProcessors returns a list of processors that should be executed during the migration process.
+// Which processors are returned depends on the current migration step indicated by the "api-gateway.kyma-project.io/migration-step" APIRule annotation.
+func NewMigrationProcessors(apiRuleV2alpha1 *gatewayv2alpha1.APIRule, apiRuleV1beta1 *gatewayv1beta1.APIRule, config processing.ReconciliationConfig, log *logr.Logger) []processing.ReconciliationProcessor {
+	step := nextMigrationStep(apiRuleV2alpha1)
+	var processors []processing.ReconciliationProcessor
```

Comment:
Maybe some logging information for debugging would be helpful, e.g. "Migrating APIRule with step %s"?